src/main/webapp/app/exam/manage/exams/exam-update.component.ts

```diff
@@ -1,142 +1,223 @@
-            } else if (!this.exam.numberOfCorrectionRoundsInExam) {
-                this.exam.numberOfCorrectionRoundsInExam = 1;
-            }
-        });
-        // Initialize helper attributes
-        this.workingTimeInMinutes = this.exam.workingTime! / 60;
-        this.calculateMaxWorkingTime();
+    ngOnDestroy() {
+        this.componentActive = false;
+    }
+
+    /**
+     * Sets the exam working time in minutes.
+     * @param minutes
+     */
+    set workingTimeInMinutes(minutes: number) {
+        this.exam.workingTime = minutes * 60;
+    }
+
+    /**
+     * Returns the exam working time in minutes.
+     */
+    get workingTimeInMinutes(): number {
+        return this.exam.workingTime ? this.exam.workingTime / 60 : 0;
+    }
+
+    get oldWorkingTime(): number | undefined {
+        return normalWorkingTime(this.originalStartDate, this.originalEndDate);
+    }
+
+    get newWorkingTime(): number | undefined {
+        return this.exam.workingTime;
     }
 
     /**
      * Revert to the previous state, equivalent with pressing the back button on your browser
-     * Returns to the detail page if there is no previous state and we edited an existing exam
-     * Returns to the overview page if there is no previous state and we created a new exam
+     * Returns to the detail page if there is no previous state, and we edited an existing exam
+     * Returns to the overview page if there is no previous state, and we created a new exam
      */
-    previousState() {
+    resetToPreviousState() {
         this.navigationUtilService.navigateBackWithOptional(['course-management', this.course.id!.toString(), 'exams'], this.exam.id?.toString());
     }
 
+    /**
+     * Updates the working time for real exams based on the start and end dates.
+     */
+    updateExamWorkingTime() {
+        if (this.exam.testExam) return;
+
+        this.exam.workingTime = examWorkingTime(this.exam);
+    }
+
+    /**
+     * Returns the maximum working time in minutes for test exams.
+     */
+    get maxWorkingTimeInMinutes(): number {
+        if (!this.exam.testExam) return 0;
+
+        if (this.exam.startDate && this.exam.endDate) {
+            return dayjs(this.exam.endDate).diff(this.exam.startDate, 'm');
+        } else {
+            // In case of an import, the exam.workingTime is imported, but the start / end date are deleted -> no error should be shown to the user in this case
+            return this.isImport ? this.workingTimeInMinutes : 0;
+        }
+    }
+
+    /**
+     * Saves the exam. If the dates have changed and the exam is ongoing, a confirmation modal is shown to the user.
+     * If either the user confirms the modal, the exam is not ongoing or the dates have not changed, the exam is saved.
+     */
+    handleSubmit() {
+        const datesChanged = this.exam.startDate?.diff(this.originalStartDate) !== 0 || this.exam.endDate?.diff(this.originalEndDate) !== 0;
```

Comment:
Can't you just use .equals() or something?

Reply:
Changed to `isSame`